src/box/lua/schema.lua

```diff
@@ -2525,6 +2525,9 @@ end
 -- the same tuple as index:pairs(key, {iterator = it})
 base_index_mt.offset_of = function(index, key, opts)
     check_index_arg(index, 'offset_of', 2)
+    key = keify(key)
+    opts = opts or {}
+    check_param_table(opts, {iterator = 'string, number'}, 2)
```

Comment:
Already approved the PR, but then joined a live discussion. We use `check_param_table` only on DDL (probably, for the sake of performance). Let's not use it here, just as in other DML ops?

(I was the one who suggested to use it - sorry 😕)

Reply:
Removed the opts check.

I don't think we should check for any type except for `nil` now: using other types violates the function interface. So the only `opts = opts or {}` should be sufficient as a valid use case check.

Reply:
We can do something like
```diff
diff --git a/src/box/lua/schema.lua b/src/box/lua/schema.lua
index 2555998125..e395b5fa6c 100644
--- a/src/box/lua/schema.lua
+++ b/src/box/lua/schema.lua
@@ -2525,13 +2525,15 @@ end
 -- the same tuple as index:pairs(key, {iterator = it})
 base_index_mt.offset_of = function(index, key, opts)
     check_index_arg(index, 'offset_of', 2)
+    key = keify(key)
+    opts = opts or {}
     local itype = check_iterator_type(opts, #key == 0, 2)
     if itype == box.index.EQ then
-        opts.iterator = box.index.GE
+        itype = box.index.GE
     elseif itype == box.index.REQ then
-        opts.iterator = box.index.LE
+        itype = box.index.LE
     end
-    return index:len() - index:count(key, opts)
+    return index:len() - index:count(key, itype)
 end
 
 base_index_mt.get_ffi = function(index, key)
```
No extra checks, and it works in the same way as count. But without it:
```lua
tarantool> box.space._space:count(nil, 'EQ')
---
- 27
...

tarantool> box.space._space:offset_of(nil, 'EQ')
---
- error: 'builtin/box/schema.lua:2532: attempt to index local ''opts'' (a string value)'
...
```

Reply:
Nice solution, applied.

Reply:
No objections anymore, thanks for the fixes!